File: Hash Table/2423. Remove Letter To Equalize Frequency/equalFrequency.c
bool equalFrequency(char* word) {
    int digit[26] = {0};
    int len = strlen(word);
    for (int i = 0; i < len; i++) {
        digit[word[i] - 'a']++;
    }
    
    for (int i = 0; i < len; i++) {
        int index = word[i] - 'a';
        digit[index]--;
        int freqs = 0;
        bool temp = true;
        for (int j = 0; j < 26; j++) {
            if (digit[j] != 0) {
                if (freqs == 0) {
                    freqs = digit[j];
                } else if (freqs != digit[j]) {
                    temp = false;
                }
            }
        }
        if (temp) {
            return true;
        }
        digit[index]++;
    }
    
    return false;
}

File: Hash Table/2423. Remove Letter To Equalize Frequency/equalFrequency.py
class Solution:
    def equalFrequency(self, word: str) -> bool:
        digit = [0] * 26
        for i in range(len(word)):
            digit[ord(word[i]) - ord('a')] += 1
        
        for i in range(len(word)):
            index = ord(word[i]) - ord('a')
            digit[index] -= 1
            freqs = 0
            temp = True
            for j in range(26):
                if digit[j] != 0:
                    if freqs == 0:
                        freqs = digit[j]
                    elif freqs != digit[j]:
                        temp = False
                        break
            
            if temp:
                return True
            
            digit[index] += 1
        
        return False